### DISCUSSION TITLE
Complete the first field but doesn't focus to the next field

### DISCUSSION THREAD

**Discussion by hungify2022** [react-hook-form/react-hook-form#10297] [Q&A] [answered]
Hi all :hibiscus: 

Can someone explain to me what is different between A1 and A2
Why does only one A1 work but A2 doesn't?

Note: I got the error `Uncaught (in promise) TypeError: elm.focus is not a function` when completing the first field and then entering submit 

Detail: 
In the first field: fill something => enter
In the second field: Fill something => enter => caught error

Codesanbox: https://stackblitz.com/edit/github-hpwy8p?file=src%2FApp.tsx
Repo: https://github.com/hungify2022/rhf-focus-error

A1
```jsx
 <Controller
        control={control}
        name="cardNumber"
        render={({ field: { ref, ...restField } }) => {
          return (
            <InputMask
              {...restField}
              mask="1-1-1"
              maskChar={null}
              alwaysShowMask={false}
              inputRef={ref}
            />
          );
        }}
      />
```
vs
 
 A2
```jsx
 <Controller
        control={control}
        name="cardNumber"
        render={({ field }) => {
          return (
            <InputMask
              {...field}
              mask="1-1-1"
              maskChar={null}
              alwaysShowMask={false}
              inputRef={field.ref}
            />
          );
        }}
      />
```

**Comment by fahadsohail482** [comment]
@hungify2022 can you provide the code sandbox link?

**Reply by hungify2022** [reply]
Okay. 
I just update my post.

**Comment by leapful** [comment, marked as answer]
@hungify2022 

The reason is because when using the whole `field` from `render={({ field }) => {}`, it also passed `ref` prop to `InputMask` component and causes conflict on getting DOM element for focusing by react-hook-form.

For example, it will look like this:

```
const field = { ref };

<InputMask {...field} inputRef={field.ref} />

equivalent to

<InputMask ref={field.ref} inputRef={field.ref} />
```

**Reply by hungify2022** [reply]
Aha
I got it
Thanks for your reply :100:
